packages/core-contracts/src/contracts/FleetProxy.sol

```diff
@@ -0,0 +1,334 @@
+     * @param _accessManager Address of the access manager
+     * @param _bridgeRouter Address of the bridge router
+     * @param _bridgeQueue Address of the bridge queue
+     * @param _crossChainRegistry Address of the CrossChainRegistry contract
+     * @param _fleetContract Address of the Fleet contract this proxy covers
+     */
+    constructor(
+        address _accessManager,
+        address _bridgeRouter,
+        address _bridgeQueue,
+        address _crossChainRegistry,
+        address _fleetContract
+    ) ProtocolAccessManaged(_accessManager) {
+        if (_bridgeRouter == address(0)) revert InvalidBridgeRouter();
+        if (_bridgeQueue == address(0)) revert InvalidBridgeQueue();
+        if (_crossChainRegistry == address(0)) revert InvalidRegistry();
+        if (_fleetContract == address(0)) revert InvalidFleetContract();
+
+        bridgeRouter = IBridgeRouter(_bridgeRouter);
+        bridgeQueue = IBridgeQueue(_bridgeQueue);
+        crossChainRegistry = ICrossChainRegistry(_crossChainRegistry);
+        fleetContract = _fleetContract;
+    }
+
+    /*//////////////////////////////////////////////////////////////
+                        EXTERNAL FUNCTIONS
+    //////////////////////////////////////////////////////////////*/
+
+    /// @inheritdoc IFleetProxy
+    function getBalance(address token) external view returns (uint256) {
+        return IERC20(token).balanceOf(address(this));
+    }
+
+    /// @inheritdoc IFleetProxy
+    function totalAssets() external view returns (uint256) {
+        return
+            IFleetCommander(fleetContract).totalAssets() + inflightWithdrawals;
+    }
+
+    /// @inheritdoc IFleetProxy
+    function pause() external onlyGuardian {
+        _pause();
+    }
+
+    /// @inheritdoc IFleetProxy
+    function unpause() external onlyGovernor {
+        _unpause();
+    }
+
+    /// @notice Force update the inflight withdrawals amount (emergency governance function)
+    /// @param amount Amount of withdrawal assets to set as in-flight
+    /// @dev This is an emergency function that allows governance to manually correct inflight withdrawal tracking
+    /// in case of bridge failures or accounting discrepancies
+    function forceUpdateInflightAssets(uint256 amount) external onlyGovernor {
+        inflightWithdrawals = amount;
+        emit InflightAssetsUpdated(amount);
+    }
+
+    /// @inheritdoc IInflightAssetTracking
+    function updateInflightAssets(uint256 amount) external {
+        // Only the bridge queue or router should be able to call this
+        if (
+            msg.sender != address(bridgeQueue) &&
+            msg.sender != address(bridgeRouter)
+        ) {
+            revert Unauthorized();
+        }
+
+        inflightWithdrawals = amount;
+        emit InflightAssetsUpdated(amount);
+    }
+
+    /// @notice Keeper function to withdraw and transfer assets
+    function withdrawAndTransfer(
+        uint256 amount,
```

Comment:
keeper wont be able to withdraw all of the funds since there will be someinterest accrued between feteching the amount and and submitting the tx. Ether redeem has to be used or `Constants.MAX_UINT256`

with `Constants.MAX_UINT256` the `amont` later on in the execution wont match the actual withdrawn amount and the bridge will fail.